ddcommon/src/entity_id/unix/cgroup_inode.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides functions to fetch cgroup node path and fetching it's inode
+use super::CgroupFileParsingError;
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::os::unix::fs::MetadataExt;
+use std::path::{Path, PathBuf};
+use std::{fs, io};
+
+// Those two variables are unused in tests
+#[cfg(not(test))]
+// From https://github.com/torvalds/linux/blob/5859a2b1991101d6b978f3feb5325dad39421f29/include/linux/proc_ns.h#L41-L49
+// Currently, host namespace inode number are hardcoded, which can be used to detect
+// if we're running in host namespace or not (does not work when running in DinD)
+const HOST_CGROUP_NAMESPACE_INODE: u64 = 0xEFFFFFFB;
+
+#[cfg(not(test))]
+const DEFAULT_CGROUP_NS_PATH: &str = "/proc/self/ns/cgroup";
+
+/// Returns the inode of file at `path`
+fn get_inode(path: &Path) -> io::Result<u64> {
+    let meta = fs::metadata(path)?;
+    Ok(meta.ino())
+}
+
+/// Returns the cgroup mount path associated with `base_controller` or the default one for
+/// cgroupV2
+fn get_cgroup_node_path(
+    base_controller: &str,
```

Comment:
IMO it's worth renaming (and propagating) the variable as `cgroup_v1_base_controller`